content/post/update-new-python-api.md

```diff
@@ -0,0 +1,160 @@
++++
+title = "How to update your Python applications to the new API"
+slug = "update-new-python-api"
+draft = false
+date = 2017-11-09T00:00:00Z
+tags = [
+    "wallaroo",
+    "python",
+    "how-to"
+]
+categories = [
+    "API"
+]
+description = "A how-to guide to updating to our new Python API, introduced in version 0.4.0"
+author = "amosca"
++++
+
+Thanks to the extensive feedback we have received on our first Python API, we have decided to release some improvements. So in version 0.4.0, we released some breaking changes to the Python API. Whilst these result in smaller applications with less boiler-plate code, it means you will have to update the code for your applications. To alleviate the pain, we created this blog entry to help guide you through the process. We'll be using word count as an example, but you should be able to follow along with your own code. If you have any technical questions that this post didn't answer, or if you have any suggestions, please get in touch via [our mailing list][mailing list] or [our IRC channel][irc channel]. And if you're one of the many early adopters that made suggestions on what to improve, this is our way of thanking you for the valuable feedback: we listened!
+
+
+## What's changed
+
+A few things are now different (we hope you'll agree that this is better):
+
+* We use a lot less "helper" classes, and replaced them with decorators
+* You no longer need to create StateBuilder classes
+* Header decoding is much simpler
+
+If you want to compare the two versions of the application, look at: [old file][https://github.com/WallarooLabs/wallaroo/blob/36296bcd583578f5230b6c99e0554afcd3541dc0/examples/python/word_count/word_count.py] [new file][https://github.com/WallarooLabs/wallaroo/blob/c346c0427850990af773df4c9f0cf1d56bde402d/examples/python/word_count/word_count.py].
```

Comment:
Instead of "the application", you should call out again that this is the old API version of word count and the new API version of word count